src/services/auth.js

```diff
@@ -0,0 +1,25 @@
+import React, { useContext, useState, useEffect } from "react";
+import { onAuthStateChanged } from "firebase/auth";
+import { auth } from "./firebase";
+
+const AuthContext = React.createContext();
+export const useAuth = () => useContext(AuthContext);
```

Comment:
slightly more convenient pattern so that you can just import `useAuth` instead of `useContext` and the relevant context